lib/Model/Importer/DebugInfo/DwarfImporter.cpp

```diff
@@ -1358,11 +1384,35 @@ inline void detectAliases(const llvm::object::ObjectFile &ELF,
         }
       }
 
+      // Check if we should add an ExportedName for local Functions.
+      llvm::SmallVector<std::string, 4> PotentialExportedNamesToBeAdded;
+      bool IsLocalFunction = false;
+      model::Function *TheFunction = nullptr;
+      for (auto &Name : CurrentAliases) {
+        auto It = FunctionsByName.find(Name);
+        PotentialExportedNamesToBeAdded.push_back(Name);
+        if (It != FunctionsByName.end()) {
+          // We found a local function.
+          // TODO: In some situations OriginalName is not in the ExportedNames?
+          // For example in the case of importing `__libc_calloc` from
+          // libc.so.6.
+          TheFunction = It->second;
+        }
+      }
+      // It is a local function. Populate the ExportedNames.
+      if (TheFunction) {
+        for (auto &Name : PotentialExportedNamesToBeAdded)
+          TheFunction->ExportedNames().insert(Name);
+        continue;
+      }
+
+      // Consider it as a Dynamic function.
       if (Prototype.isValid()) {
         for (const std::string &Name : UnprototypedFunctionsNames) {
           auto It = ImportedDynamicFunctions.find(Name);
           if (It == ImportedDynamicFunctions.end())
             It = ImportedDynamicFunctions.insert({ Name }).first;
+
           It->Prototype() = Prototype;
         }
       }
```

Comment:
I'm confused by this piece of code. We're iterating over symbols that *have* an address (enforced in `computeEquivalentSymbols`) but we're created new `ImportedDynamicFunctions`. This doesn't sound right at all.
I think all of this function should not touch `ImportedDynamicFunctions` at all.

We have, one non-null address, many names.
So we should:

1. Check if `Binary.Functions` contains the address, if not create it.
2. Assign to such `model::Function` the prototype.
3. If the symbol is dynamic (`computeEquivalentSymbols` only considers static symbols AFAIU), we should add the name of the dynamic symbol to `ExportedNames`.

I think this basically needs to be rewritten.

Reply:
I think that we should generate ```ExportedNames``` for local functions only. The rest of the comment makes perfectly sense to me. I've rewritten the code.

Reply:
We need to add `ExportedNames` for `model::Function` but only for names coming from dynamic symbols.
If we have an alias in `.symtab` it's good to know, but, unless it's also in `.dynsym`, it's not an "exported name", you can't refer to a symbol with a name that's not in `.dynsym`.